go-controller/cmd/ovnkube/ovnkube.go

```diff
@@ -137,81 +233,105 @@ func main() {
 		if err != nil {
 			panic(err.Error())
 		}
-		clusterController.SouthDBClientAuth, err = ovncluster.NewOvnDBAuth(*ovnSouth, *ovnSouthClientPrivKey, *ovnSouthClientCert, *ovnSouthClientCACert, false)
+
+		ovnSouth := ctx.String("ovn-south-db")
+		ovnSouthClientPrivKey := ctx.String("ovn-south-client-privkey")
+		ovnSouthClientCert := ctx.String("ovn-south-client-cert")
+		ovnSouthClientCACert := ctx.String("ovn-south-client-cacert")
+		clusterController.SouthDBClientAuth, err = ovncluster.NewOvnDBAuth(ovnSouth, ovnSouthClientPrivKey, ovnSouthClientCert, ovnSouthClientCACert, false)
 		if err != nil {
 			panic(err.Error())
 		}
 
-		if *node != "" {
-			if *token == "" {
+		if node != "" {
+			if token == "" {
 				panic("Cannot initialize node without service account 'token'. Please provide one with --token argument")
 			}
 
-			err := clusterController.StartClusterNode(*node)
+			err := clusterController.StartClusterNode(node)
 			if err != nil {
 				logrus.Errorf(err.Error())
 				panic(err.Error())
 			}
 		}
-		clusterController.NodePortEnable = *nodePortEnable
+		clusterController.NodePortEnable = nodePortEnable
 
-		if *master != "" {
+		if master != "" {
 			if runtime.GOOS == "windows" {
 				panic("Windows is not supported as master node")
 			}
-			clusterController.NorthDBServerAuth, err = ovncluster.NewOvnDBAuth(*ovnNorth, *ovnNorthServerPrivKey, *ovnNorthServerCert, *ovnNorthServerCACert, true)
+			ovnNorthServerPrivKey := ctx.String("ovn-north-server-privkey")
+			ovnNorthServerCert := ctx.String("ovn-north-server-cert")
+			ovnNorthServerCACert := ctx.String("ovn-north-server-cacert")
+			clusterController.NorthDBServerAuth, err = ovncluster.NewOvnDBAuth(ovnNorth, ovnNorthServerPrivKey, ovnNorthServerCert, ovnNorthServerCACert, true)
 			if err != nil {
 				panic(err.Error())
 			}
-			clusterController.SouthDBServerAuth, err = ovncluster.NewOvnDBAuth(*ovnSouth, *ovnSouthServerPrivKey, *ovnSouthServerCert, *ovnSouthServerCACert, true)
+
+			ovnSouthServerPrivKey := ctx.String("ovn-south-server-privkey")
+			ovnSouthServerCert := ctx.String("ovn-south-server-cert")
+			ovnSouthServerCACert := ctx.String("ovn-south-server-cacert")
+			clusterController.SouthDBServerAuth, err = ovncluster.NewOvnDBAuth(ovnSouth, ovnSouthServerPrivKey, ovnSouthServerCert, ovnSouthServerCACert, true)
 			if err != nil {
 				panic(err.Error())
 			}
 
 			// run the cluster controller to init the master
-			err := clusterController.StartClusterMaster(*master)
+			err := clusterController.StartClusterMaster(master)
 			if err != nil {
 				logrus.Errorf(err.Error())
 				panic(err.Error())
 			}
 		}
 	}
-	if *netController {
+	if netController {
 		ovnController := ovn.NewOvnController(clientset, factory)
-		ovnController.NodePortEnable = *nodePortEnable
+		ovnController.NodePortEnable = nodePortEnable
 		ovnController.Run()
 	}
-	if *master != "" || *netController {
+	if master != "" || netController {
```

Comment:
This is unrelated to this PR.

Why do we run forever if we do only a master-init? Shouldn't it run forever only if the netController is specified as well? Like the following:
```
master != "" && netController
```

At the moment ovnkube returns only if called without master name and netController flag. In this case it does almost nothing useful. In my opinion there should be a warning at least.

Reply:
@alinbalutoiu in the 'master' and netController roles, it needs to run forever.  The master code has the node watch to allocate CIDRs to nodes, and the netController handles policy and such.  They both need to continue running.

At the moment, only the node can be run & quit.

Reply:
Thanks a lot for explaining it!